apps/zotonic_core/src/models/m_rsc_import.erl

```diff
@@ -18,96 +17,1130 @@
+                {Acc#{ <<"blocks">> => Blocks1 }, ImpAcc1};
+            (K, V, {Acc, ImpAcc}) ->
+                case m_rsc_export:is_id_prop(K) of
+                    true ->
+                        case map_id(V, UriTemplate, ImpAcc, OptionsNoEdges, Context) of
+                            {ok, {LocalId, ImpAcc1}} ->
+                                ImpAcc2 = ImpAcc1#{
+                                    uri(V) => LocalId
+                                },
+                                {Acc#{ K => LocalId }, ImpAcc2};
+                            {error, _} ->
+                                {Acc#{ K => undefined }, ImpAcc}
+                        end;
+                    false ->
+                        {V1, ImpAcc1} = map_html(K, V, UriTemplate, ImpAcc, OptionsNoEdges, Context),
+                        {Acc#{ K => V1 }, ImpAcc1}
+                end
+        end,
+        {#{}, ImportedAcc},
+        Rsc),
+
+    % Set forced and default props
+    Rsc2 = maps:merge(Rsc1, PropsForced),
+    Rsc3 = maps:merge(PropsDefault, Rsc2),
+    Rsc4 = case maps:find(<<"is_authoritative">>, PropsForced) of
+        {ok, true} ->
+            Rsc3#{
+                <<"is_authoritative">> => true,
+                <<"uri">> => undefined
+            };
+        _ ->
+            Rsc3#{
+                <<"uri">> => maps:get(<<"uri">>, RemoteRId),
+                <<"is_authoritative">> => false
+            }
+    end,
+
+    % Ensure that the is_published flag is present, defaults to true
+    Rsc5 = case maps:find(<<"is_published">>, Rsc4) of
+        {ok, true} -> Rsc4;
+        _ -> Rsc4#{ <<"is_published">> => true }
+    end,
+    {Rsc5, ImportedAcc1}.
+
+
+%% @doc Map all ids in a menu to local (stub) resources. Skip all tree entries
+%% that could not be mapped to local ids.
+map_menu([ #rsc_tree{ id = RemoteId, tree = Sub } | Rest ], UriTemplate, Acc, ImpAcc, Options, Context) ->
+    case map_id(RemoteId, UriTemplate, ImpAcc, Options, Context) of
+        {ok, {LocalId, ImpAcc1}} ->
+            {Sub1, ImpAcc2} = map_menu(Sub, UriTemplate, [], ImpAcc1, Options, Context),
+            Acc1 = [ #rsc_tree{ id = LocalId, tree = Sub1 } | Acc ],
+            map_menu(Rest, UriTemplate, Acc1, ImpAcc2, Options, Context);
+        {error, _} ->
+            % Skip the tree entry if the remote id could not be mapped
+            map_menu(Rest, UriTemplate, Acc, ImpAcc, Options, Context)
+    end;
+map_menu([ _ | Rest ], UriTemplate, Acc, ImpAcc, Options, Context) ->
+    map_menu(Rest, UriTemplate, Acc, ImpAcc, Options, Context);
+map_menu([], _UriTemplate, Acc, ImpAcc, _Options, _Context) ->
+    {lists:reverse(Acc), ImpAcc}.
+
+
+%% @doc Map all ids in blocks to local (stub) resources. Remove blocks that can not
+%% have their ids mapped.
+map_blocks([ B | Rest ], UriTemplate, Acc, ImpAcc, Options, Context) ->
+    OptionsNoEdges = proplists:delete(import_edges, Options),
```

Comment:
I am not sure I understand why it is called `OptionsNoEdges` while the value holds what is in `import_edges` - the tandem implies a boolean, but I'm not sure how the semantics of `import_edges = true` would match the naming of `OptionsNoEdges`; I'd guess the contrary.

Reply:
Yes, better would be `OptionsWithoutEdges`

Reply:
Changed it.